test/new-e2e/tests/windows/install-test/system_file_tester.go

```diff
@@ -0,0 +1,199 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package installtest
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/testutil/flake"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/components"
+
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// SystemFileIntegrityTester is a test helper for testing that system files are not removed.
+type SystemFileIntegrityTester struct {
+	host *components.RemoteHost
+
+	firstSnapshotPath  string
+	secondSnapshotPath string
+}
+
+// NewSystemFileIntegrityTester creates a new SystemFileIntegrityTester
+func NewSystemFileIntegrityTester(host *components.RemoteHost) *SystemFileIntegrityTester {
+	t := &SystemFileIntegrityTester{}
+
+	t.host = host
+
+	t.firstSnapshotPath = `C:\Windows\SystemDirListBeforeInstall.txt`
+	t.secondSnapshotPath = `C:\Windows\SystemDirListAfterUninstall.txt`
+
+	return t
+}
+
+// FirstSnapshotTaken returns true if the first snapshot exists
+func (t *SystemFileIntegrityTester) FirstSnapshotTaken() (bool, error) {
+	return t.host.FileExists(t.firstSnapshotPath)
+}
+
+// TakeSnapshot takes a snapshot of the system files that can be used to compare against later.
+// The snapshot is overridden if it already exists.
+func (t *SystemFileIntegrityTester) TakeSnapshot() error {
+	return snapshotSystemfiles(t.host, t.firstSnapshotPath)
+}
+
+// RemoveSnapshots removes any snapshots if they exist
+func (t *SystemFileIntegrityTester) RemoveSnapshots() error {
+	exists, err := t.host.FileExists(t.firstSnapshotPath)
+	if err != nil {
+		return fmt.Errorf("failed to check if first snapshot exists: %w", err)
+	}
+	if exists {
+		err := t.host.Remove(t.firstSnapshotPath)
+		if err != nil {
+			return fmt.Errorf("failed to remove first snapshot: %w", err)
+		}
+	}
+	exists, err = t.host.FileExists(t.secondSnapshotPath)
+	if err != nil {
+		return fmt.Errorf("failed to check if second snapshot exists: %w", err)
+	}
+	if exists {
+		err := t.host.Remove(t.secondSnapshotPath)
+		if err != nil {
+			return fmt.Errorf("failed to remove second snapshot: %w", err)
+		}
+	}
+	return nil
```

Comment:
Not sure if this is a problem - if we fail to remove the first snapshot, then we don't bother looking for or removing the second.  Is that intended? I suspect we'd want to try and remove what we can, but let me know why you thought it might be better to not try to get rid of the second, if we can't get rid of the first.